var/spack/repos/builtin/packages/dune-common/package.py

```diff
@@ -0,0 +1,27 @@
+# copyright 2013-2023 lawrence livermore national security, llc and other
+# spack project developers. see the top-level copyright file for details.
+#
+# spdx-license-identifier: (apache-2.0 or mit)
+
+
+from spack.package import *
+
+
+class dunecommon(pythonpackage):
+    """dune, the distributed and unified numerics environment
+        is a modular toolbox for solving partial differential
+        equations (pdes) with grid-based methods.
+        it supports the easy implementation of methods
+        like finite elements (fe), finite volumes (fv),
+        and also finite differences (fd).
+    """
+
+    homepage = "https://www.dune-project.org/doc/gettingstarted/"
+    git = "https://gitlab.dune-project.org/core/dune-common"
+    pypi = "dune-common/dune-common-2.9.0.tar.gz"
+
+    version("2.9.0")
```

Comment:
Add a `sha256` argument here; otherwise, Spack cannot fetch the package.  (Use `spack checksum dune-common 2.9.0` to get a formatted directive that includes the sha256.)

```
==> Error: Attempt to check URLFetchStrategy with no digest.
```

See https://spack.readthedocs.io/en/latest/packaging_guide.html#spack-checksum.